data-prepper-expression/src/main/antlr/DataPrepperStatement.g4

```diff
@@ -0,0 +1,262 @@
+    ;
+
+fragment
+JsonPointerEscapeSequence
+    : '\\' [btnfr"'\\/]
+    ;
+
+fragment
+JsonPointerStringCharacters
+    : JsonPointerStringCharacter+
+    ;
+
+fragment
+JsonPointerStringCharacter
+    : ~["\\]
+    | JsonPointerEscapeSequence
+    ;
+
+fragment
+JsonPointerCharacter
+    : [A-Za-z0-9_]
+    ;
+
+fragment
+JsonPointerCharacters
+    : JsonPointerCharacter+
+    ;
+
+JsonPointer
+    : DOUBLEQUOTE FORWARDSLASH JsonPointerStringCharacters? DOUBLEQUOTE
+    | FORWARDSLASH JsonPointerCharacters? (FORWARDSLASH JsonPointerCharacters)*
+    ;
+
+String
+    : DOUBLEQUOTE StringCharacters? DOUBLEQUOTE
+    ;
+
+fragment
+VariableNameLeadingCharacter
+    : [A-Za-z_]
+    ;
+
+fragment
+VariableNameCharacter
+    : VariableNameLeadingCharacter
+    | [0-9-]
+    ;
+
+fragment
+VariableNameCharacters
+    : VariableNameLeadingCharacter VariableNameCharacter*
+    ;
+
+VariableIdentifier
+    : '${' VariableNameCharacters '}'
+    ;
+
+statement
+    : expression EOF
+    | OTHER {System.err.println("unknown char: " + $OTHER.text);}
+    ;
+
+expression
+    : binaryOperatorExpression
+    ;
+
+binaryOperatorExpression
+    : conditionalExpression
+    ;
+
+conditionalExpression
+    : conditionalExpression conditionalOperator equalityOperatorExpression
+    | equalityOperatorExpression
+    ;
+
+equalityOperatorExpression
+    : equalityOperatorExpression binaryOperator regexOperatorExpression
+    | regexOperatorExpression
+    ;
+
+regexOperatorExpression
+    : regexOperatorExpression regexEqualityOperator regexPattern
+    | relationalOperatorExpression
+    ;
+
+relationalOperatorExpression
+    : relationalOperatorExpression relationalOperator setOperatorExpression
+    | setOperatorExpression
+    ;
+
+setOperatorExpression
+    : setOperatorExpression setOperator setInitializer
+    | unaryOperatorExpression
+    ;
+
+unaryOperatorExpression
+    : primary
+    | setInitializer
+    | regexPattern
+    | unaryNotOperatorExpression
+    ;
+
+unaryNotOperatorExpression
+    : unaryOperator primary
+    ;
+
+binaryOperator
+    : relationalOperator
+    | equalityOperator
+    ;
+
+regexEqualityOperator
+    : '=~'
+    | '!~'
+    ;
+
+setOperator
+    : 'in'
+    | 'not in'
+    ;
+
+
+conditionalOperator
+    : 'and'
+    | 'or'
+    ;
+
+unaryOperator
+    : 'not'
+    ;
+
+equalityOperator
+    : '=='
+    | '!='
+    ;
+
+relationalOperator
+    : '<'
+    | '<='
+    | '>'
+    | '>='
+    ;
+
+primary
+    : literal
+    | variableIdentifier
+    | setInitializer
+    | expressionInitializer
+    ;
+
+regexPattern
+    : JsonPointer
+    | String
+    | '(' (JsonPointer | String) ')'
+    ;
+
+expressionInitializer
+    : '(' expression ')'
+    ;
+
+setInitializer
+    : '{' binaryOperatorExpression (',' binaryOperatorExpression)* '}'
+    ;
+
+variableIdentifier
+    : variableName
+    ;
+
+variableName
+    : VariableIdentifier
+    ;
+
+literal
+    : Float
+    | Integer
+    | Boolean
+    | JsonPointer
```

Comment:
The RFC did call for making `JsonPointer` a `literal`. Should it be a literal, or should primary be able to take a `JsonPointer` instead?

```
primary
     : literal
     | JsonPointer
     | variableIdentifier
     | setInitializer
     | expressionInitializer
     ;
```